app-typescript/components/Lists/TableList.tsx

```diff
@@ -283,85 +286,90 @@ class TableListItem extends React.PureComponent<IPropsItem> {
     }
 
     render() {
-
         let classes = classNames('table-list__item', {
             'table-list__item--clickable': this.props.onClick,
             'table-list__item--draggable': this.props.dragAndDrop,
             'table-list__item--locked': this.props.locked,
             'table-list__item--position-locked': this.props.positionLocked,
             'table-list__item--drag-handles-always': !this.props.showDragHandle,
             'table-list__item--drag-handles-none': this.props.showDragHandle === 'none',
+            'table-list__item--margin': !this.props.addItem,
         });
 
+        const Wrapper: React.ComponentType<{children: JSX.Element}> = this.props.addItem
+            ? ({children}) => (<div className='table-list__item-container'>{children}</div>)
+            : ({children}) => children;
+
         return (
-            this.props.addItem ?
-                <div className='table-list__item-container'>
+            <Wrapper>
+                <>
                     <div
                         role='listitem'
-                        onClick={() => this.onSingleClick()}
-                        onDoubleClick={() => this.onDoubleClick()}
-                        className={classes}>
+                        className={classes}
+                        onClick={(e) => this.multiClickHandler(e, {
+                            onSingleClick: () => {
+                                let selection = window.getSelection();
+                                if (this.props.onClick && selection) {
+                                    if (this.props.dragAndDrop) {
+                                        this.props.onClick();
+                                    } else {
+                                        if (selection.toString().length < 1) {
+                                            this.props.onClick();
+                                        }
+                                    }
+                                }
+                            },
+                            onDoubleClick: () => {
+                                if (this.props.onDoubleClick) {
+                                    this.props.onDoubleClick();
+                                }
+                            },
+                        })}
+                    >
                         <div className='table-list__item-border' style={{ backgroundColor: this.props.hexColor }}></div>
                         <div className='table-list__item-content'>
                             <div className='table-list__item-content-block'>
                                 {this.props.start && this.props.start}
                             </div>
-                            <div className='table-list__item-content-block table-list__item-content-block--center'>
+                            <div className='table-list__item-content-block table-list__item-content-block--center table-list__item-content-block--title'>
```

Comment:
is `table-list__item-content-block--title` used anywhere?

Reply:
nope... ill remove this place...